scripts-available/CDB_Helper.sql

```diff
@@ -158,3 +158,21 @@ BEGIN
   RETURN left(string, (i - 1));
 END;
 $$ LANGUAGE 'plpgsql' IMMUTABLE PARALLEL SAFE;
+
+
+-- Checks if a given text representing a qualified or unqualified table name (relation)
+-- actually exists in the database. It is meant to be used as a guard for other function/queries.
+CREATE OR REPLACE FUNCTION cartodb._CDB_Table_Exists(table_name_with_optional_schema TEXT)
+RETURNS bool
+AS $$
+BEGIN
+    IF EXISTS(SELECT * FROM pg_class WHERE table_name_with_optional_schema::regclass::oid = oid AND relkind = 'r') THEN
```

Comment:
Can this be done without the `if`? That is assigning it to a variable and returning it plus catching the exception. I'm asking without knowing if its possible in plpgsql but it feels more natural.

Reply:
Actually I think that's the recommended construction for it, so that no extra variable (plus check) is needed.